src/Process/Listener/MutationTestingResultsLoggerSubscriber.php

```diff
@@ -72,54 +82,63 @@ public function onMutationTestingFinished(MutationTestingFinished $event)
 
         $logTypes = $this->filterLogTypes($logTypes);
 
-        foreach ($logTypes as $logType => $directory) {
-            $this->useLogger($logType);
+        foreach ($logTypes as $logType => $config) {
+            $this->useLogger($logType, $config);
         }
     }
 
     private function filterLogTypes(array $logTypes): array
     {
         $allowedFileTypes = [
-            self::TEXT_FILE,
-            self::DEBUG_FILE,
-            self::SUMMARY_FILE,
+            ResultsLoggerTypes::TEXT_FILE,
+            ResultsLoggerTypes::DEBUG_FILE,
+            ResultsLoggerTypes::SUMMARY_FILE,
+            ResultsLoggerTypes::BADGE,
         ];
 
         foreach ($logTypes as $key => $value) {
-            if (!in_array($key, $allowedFileTypes)) {
+            if (!in_array($key, $allowedFileTypes, true)) {
                 unset($logTypes[$key]);
             }
         }
 
         return $logTypes;
     }
 
-    private function useLogger(string $logType)
+    private function useLogger(string $logType, $config)
```

Comment:
is the second parameter a string?

Reply:
no, each logger has its own config: somewhere it's a string, somewhere it's an object

```json
{
    "logs": {
        "text": "infection-log.txt",
        "badge": {
            "branch": "mutation-badge"
        }
    }
}